trieste/models/model_interfaces.py

```diff
@@ -29,23 +30,40 @@ class ProbabilisticModel(tf.Module, ABC):
     @abstractmethod
     def predict(self, query_points: TensorType) -> Tuple[TensorType, TensorType]:
         """
-        Return the predicted mean and variance of the latent function(s) at the specified
+        Return the mean and variance of the independent marginal distributions at each point in
         ``query_points``.
 
-        :param query_points: The points at which to make predictions.
-        :return: The predicted mean and variance.
+        This is essentially a convenience method for :meth:`predict_joint`, where non-event
+        dimensions of ``query_points`` are all interpreted as broadcasting dimensions instead of
+        batch dimensions, and the covariance is squeezed to remove redundant nesting.
+
+        :param query_points: The points at which to make predictions, of shape [..., D].
+        :return: The mean and variance of the independent marginal distributions at each point in
+            ``query_points``. For a predictive distribution with event shape E, the mean and
+            variance will both have shape [...] + E.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def predict_joint(self, query_points: TensorType) -> Tuple[TensorType, TensorType]:
```

Comment:
i'm not decided on the names yet. I'd prefer `predict` to be the main method, and `predict_independent` to be the convenience method

Reply:
as discussed, I like the current choice